service/src/main/java/uk/gov/hmcts/reform/fpl/service/legalcounsel/RepresentableLegalCounselUpdater.java

```diff
@@ -0,0 +1,186 @@
+package uk.gov.hmcts.reform.fpl.service.legalcounsel;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.reform.ccd.model.Organisation;
+import uk.gov.hmcts.reform.fpl.enums.SolicitorRole;
+import uk.gov.hmcts.reform.fpl.events.legalcounsel.LegalCounsellorRemoved;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.Child;
+import uk.gov.hmcts.reform.fpl.model.LegalCounsellor;
+import uk.gov.hmcts.reform.fpl.model.Respondent;
+import uk.gov.hmcts.reform.fpl.model.RespondentSolicitor;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.interfaces.WithSolicitor;
+import uk.gov.hmcts.reform.fpl.service.OrganisationService;
+import uk.gov.hmcts.reform.fpl.service.UserService;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.function.Predicate.not;
+import static java.util.stream.Collectors.toUnmodifiableList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.unwrapElements;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor(onConstructor_ = {@Autowired})
+public class RepresentableLegalCounselUpdater {
+
+    private final OrganisationService orgService;
+    private final UserService user;
+
+    public Set<LegalCounsellorRemoved> buildEventsForAccessRemoval(CaseData caseData, CaseData caseDataBefore,
+                                                                   final SolicitorRole.Representing representing) {
+        // need to find the representables that have been updated
+        List<WithSolicitor> current = unwrapElements(representing.getTarget().apply(caseData));
+        List<WithSolicitor> old = unwrapElements(representing.getTarget().apply(caseDataBefore));
+
+        Set<LegalCounsellor> allCurrentLegalCounsellors = Arrays.stream(SolicitorRole.Representing.values())
+            .flatMap(r -> unwrapElements(r.getTarget().apply(caseData)).stream())
+            .flatMap(s -> unwrapElements(s.getLegalCounsellors()).stream())
```

Comment:
I think it might better for readability if we didn't shorten `r` and `s`, people could then tell what they are immediately without needing to think about it